src/tools/editor.c

```diff
@@ -35,97 +36,92 @@
+    } else {
+        log_debug("[Editor] could not get JID");
+        gchar* data_dir = files_get_data_path(DIR_EDITOR);
+        if (!create_dir(data_dir)) {
             return TRUE;
         }
-        close(fd_output_file);
+        filename = g_strdup_printf("%s/compose.md", data_dir);
+        g_free(data_dir);
+    }
+    if (!filename) {
+        log_error("[Editor] something went wrong while creating compose file");
+        return TRUE;
     }
 
-    if (creation_error) {
-        cons_show_error("Editor: could not create temp file");
+    gsize messagelen = strlen(message);
+    if (!g_file_set_contents(filename, message, messagelen, &glib_error)) {
+        log_error("[Editor] could not write to %s: %s", filename, glib_error ? glib_error->message : "No GLib error given");
+        if (glib_error) {
+            g_error_free(glib_error);
+        }
+        g_free(filename);
         return TRUE;
     }
-    g_object_unref(fos);
 
     char* editor = prefs_get_string(PREF_COMPOSE_EDITOR);
 
     // Fork / exec
     pid_t pid = fork();
     if (pid == 0) {
-        int x = execlp(editor, editor, g_file_get_path(file), (char*)NULL);
+        int x = execlp(editor, editor, filename, (char*)NULL);
         if (x == -1) {
-            cons_show_error("Editor:Failed to exec %s", editor);
+            log_error("[Editor] Failed to exec %s", editor);
         }
         _exit(EXIT_FAILURE);
     } else {
         if (pid == -1) {
             return TRUE;
         }
-        int status = 0;
-        waitpid(pid, &status, 0);
-        int fd_input_file = open(g_file_get_path(file), O_RDONLY);
-        const size_t COUNT = 8192;
-        char buf[COUNT];
-        ssize_t size_read = read(fd_input_file, buf, COUNT);
-        if (size_read > 0 && size_read <= COUNT) {
-            buf[size_read - 1] = '\0';
-            GString* text = g_string_new(buf);
-            *returned_message = g_strdup(text->str);
-            g_string_free(text, TRUE);
-        } else {
-            *returned_message = g_strdup("");
-        }
-        close(fd_input_file);
+        waitpid(pid, NULL, 0);
 
-        GError* deletion_error = NULL;
-        g_file_delete(file, NULL, &deletion_error);
-        if (deletion_error) {
-            cons_show("Editor: error during file deletion");
-            g_free(*returned_message);
+        gchar* contents;
+        gsize length;
+        if (!g_file_get_contents(filename, &contents, &length, &glib_error)) {
+            log_error("[Editor] could not read from %s: %s", filename, glib_error ? glib_error->message : "No GLib error given");
+            if (glib_error) {
+                g_error_free(glib_error);
+            }
+            g_free(filename);
+            g_free(editor);
             return TRUE;
         }
-        g_object_unref(file);
+        /* Remove all trailing new-line characters */
+        while (contents[length - 1] == '\r' || contents[length - 1] == '\n') {
```

Comment:
I think we could use `g_strchomp()` here.

Reply:
nice, I forgot that this one exists